applications/callflow/src/module/cf_voicemail.erl

```diff
@@ -1408,26 +1414,30 @@ change_pin(#mailbox{mailbox_id=Id
                 Box;
             {'error', _Reason} ->
                 lager:debug("box failed validation: ~p", [_Reason]),
-                invalid_pin(Box, Call)
+                invalid_pin(Box, Call, Loop)
         end
     catch
         'error':{'badmatch',{'error','channel_hungup'}} ->
             lager:debug("channel hungup while configuring pin"),
             {'error', 'channel_hungup'};
         'error':{'badmatch',{'ok',_ConfirmPin}} ->
             lager:debug("new pin was invalid, try again"),
-            invalid_pin(Box, Call);
+            invalid_pin(Box, Call, Loop);
         _E:_R ->
             lager:debug("failed to get new pin: ~s: ~p", [_E, _R]),
-            invalid_pin(Box, Call)
+            invalid_pin(Box, Call, Loop)
     end.
 
--spec invalid_pin(mailbox(), kapps_call:call()) ->
+-spec invalid_pin(mailbox(), kapps_call:call(), non_neg_integer()) ->
                          mailbox() |
                          {'error', any()}.
-invalid_pin(Box, Call) ->
+invalid_pin(_Box, Call, Loop) when Loop > 2 -> %% third failure, terminate
```

Comment:
create a `macro` at top `MAX_CHANGE_PIN_TRIES` (or a better name) and use it here instead of  `2`

Reply:
or even better make it system configurable and add it to `mailbox` record.